docs/extensibility.md

```diff
@@ -29,531 +28,630 @@ pub trait DiscoveryHandler {
 
-Next, we need to include the gRPC generated code in by adding a reference to `nessie` in `samples/brokers/nessie/src/util/mod.rs`:
+To access the HTTP-based Device data, we first need to retrieve the discovery information.  Any information stored in the DiscoveryResult properties map will be transferred into the broker container's environment variables.  Retrieving them is simply a matter of querying environment variables like this:
 
 ```rust
-pub mod nessie;
+let device_url = env::var("AKRI_HTTP_DEVICE_ENDPOINT")?;
 ```
 
-With the gRPC implementation created, we can now start utilizing it.
-
-First, we need to leverage the generated gRPC code by creating `samples/brokers/nessie/src/util/nessie_service.rs`:
+For our HTTP broker, the data can be retrieved with a simple GET:
 
 ```rust
-use super::{
-    nessie::{
-        nessie_server::{Nessie, NessieServer},
-        NotifyRequest, NotifyResponse,
-    },
-    FrameBuffer,
-};
-use std::net::SocketAddr;
-use std::sync::{Arc, Mutex};
-use tonic::{transport::Server, Request, Response};
-
-pub const NESSIE_SERVER_ADDRESS: &str = "0.0.0.0";
-pub const NESSIE_SERVER_PORT: &str = "8083";
-
-pub struct NessieService {
-    frame_rx: Arc<Mutex<FrameBuffer>>,
-}
-
-#[tonic::async_trait]
-impl Nessie for NessieService {
-    async fn get_nessie_now(
-        &self,
-        _request: Request<NotifyRequest>,
-    ) -> Result<Response<NotifyResponse>, tonic::Status> {
-        Ok(Response::new(NotifyResponse {
-            frame: match self.frame_rx.lock().unwrap().pop_front() {
-                Some(data) => data,
-                _ => vec![],
-            },
-        }))
-    }
-}
-
-pub async fn serve(frame_rx: Arc<Mutex<FrameBuffer>>) -> Result<(), String> {
-    let nessie = NessieService { frame_rx };
-    let service = NessieServer::new(nessie);
-
-    let addr_str = format!("{}:{}", NESSIE_SERVER_ADDRESS, NESSIE_SERVER_PORT);
-    let addr: SocketAddr = match addr_str.parse() {
-        Ok(sock) => sock,
-        Err(e) => {
-            return Err(format!("Unable to parse socket: {:?}", e));
+async fn read_sensor(device_url: &str) {
+    match get(device_url).await {
+        Ok(resp) => {
+            let body = resp.text().await;
         }
+        Err(err) => println!("Error: {:?}", err),
     };
-
-    tokio::spawn(async move {
-        Server::builder()
-            .add_service(service)
-            .serve(addr)
-            .await
-            .expect("couldn't build server");
-    });
-    Ok(())
 }
 ```
 
-Once the gRPC code is utilized, we need to include our nessie server code by adding a reference to `nessie_service` in `samples/brokers/nessie/src/util/mod.rs`:
-
-```rust
-pub mod nessie_service;
-```
-
-
-Finally, we can tie all the pieces together in our main and retrieve the url from the Configuration in `samples/brokers/nessie/src/main.rs`:
+We can tie all the pieces together in our main and retrieve the url from the Configuration in `samples/brokers/http/src/main.rs`:

```

Comment:
is the url from the Configuration? Isn't if from the environment variable via the Instance?